test/hummingbot/connector/exchange/bitmart/test_bitmart_api_user_stream_data_source.py

```diff
@@ -0,0 +1,276 @@
+                                                        messages))
+            asyncio.get_event_loop().run_until_complete(self.listening_task)
+        except Exception:
+            pass
+
+        self.assertTrue(self._is_logged("ERROR", "Error occurred when authenticating to user stream."))
+        self.assertTrue(self._is_logged("ERROR", "Unexpected error with BitMart WebSocket connection. "
+                                                 "Retrying after 30 seconds..."))
+
+    @patch('websockets.connect', new_callable=AsyncMock)
+    def test_listening_process_logs_exception_during_events_subscription(self, ws_connect_mock):
+        messages = asyncio.Queue()
+        ws_connect_mock.return_value = self.mocking_assistant.create_websocket_mock()
+        ws_connect_mock.return_value.send.side_effect = lambda sent_message: (
+            self._raise_exception(Exception)
+            if "order:HBOT_USDT" in sent_message
+            else self.mocking_assistant._sent_websocket_text_messages[ws_connect_mock.return_value].append(sent_message)
+        )
+        # Make the close function raise an exception to finish the execution
+        ws_connect_mock.return_value.close.side_effect = lambda: self._raise_exception(Exception)
+
+        try:
+            self.listening_task = asyncio.get_event_loop().create_task(
+                self.data_source.listen_for_user_stream(asyncio.get_event_loop(),
+                                                        messages))
+            # Add the authentication response for the websocket
+            self.mocking_assistant.add_websocket_text_message(
+                ws_connect_mock.return_value,
+                json.dumps({"event": "login"}))
+            asyncio.get_event_loop().run_until_complete(self.listening_task)
+        except Exception:
+            pass
+
+        self.assertTrue(self._is_logged("ERROR", "Error occured during subscribing to Bitmart private channels."))
+        self.assertTrue(self._is_logged("ERROR", "Unexpected error with BitMart WebSocket connection. "
+                                                 "Retrying after 30 seconds..."))
+
+    @patch('websockets.connect', new_callable=AsyncMock)
+    def test_listening_process_timeout_inner_message(self, ws_connect_mock):
+        messages = asyncio.Queue()
+        ws_connect_mock.return_value = self.mocking_assistant.create_websocket_mock()
+
+        self.listening_task = asyncio.get_event_loop().create_task(
+            self.data_source.listen_for_user_stream(asyncio.get_event_loop(),
+                                                    messages))
+        # Add the authentication response for the websocket
+        self.mocking_assistant.add_websocket_text_message(
+            ws_connect_mock.return_value,
+            json.dumps({"event": "login"}))
+
+        self.mocking_assistant.add_websocket_text_message(ws_connect_mock.return_value, json.dumps('dummyMessage'))
```

Comment:
This test does not have any assertion. It seems to be incomplete.

Reply:
This was unintentional, the test cases are covered in the following tests, I'm removing this one.